### PR TITLE
ROX-18436: update mock-grpc-server container to be multi-arch

### PR DESCRIPTION
**Pull request by kcrane**
## Description

update mock-grpc-server to support multiple architectures.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Build and pushed to personal quay repo

### CODE REVIEW

**Comment by Stringy** [comment]
We also need to modify this GHA job: https://github.com/stackrox/stackrox/blob/aa9f55566da9e53459c8143c59566ab33e4604b4/.github/workflows/build.yaml#L586 to build all the architectures and create the relevant manifest (credit to @Molter73 for finding it)

File: Makefile [hunk 1]
```diff
@@ -645,10 +647,19 @@ syslog-image: syslog-build
...
+		-f integration-tests/mock-grpc-server/image/Dockerfile \
+		integration-tests/mock-grpc-server/image
+
+.PHONY: mock-grpc-server-image-push
+mock-grpc-server-image-push:
+	find bin -type f -name mock-grpc-server -exec install -D  "{}" "integration-tests/mock-grpc-server/image/{}" \;
+	docker buildx build --platform $(PLATFORM) --push \
+		-t quay.io/rhacs-eng/grpc-server:$(TAG) \
+		-f integration-tests/mock-grpc-server/image/Dockerfile \
 		integration-tests/mock-grpc-server/image
```
[the remark this answers is not in the document]
Comment (kcrane) [reply Makefile]: unfortunately not.  I docker can't store multi-arch images locally (see https://github.com/stackrox/collector/pull/1269#discussion_r1290731163).  So PLATFORM can only have one entry for mock-grpc-server-image.

File: Makefile [hunk 2]
```diff
@@ -645,12 +647,19 @@ syslog-image: syslog-build
...
+	cp -R bin integration-tests/mock-grpc-server/image
+	docker buildx build --platform $(PLATFORM) --load \
 		-t stackrox/grpc-server:$(TAG) \
 		-t quay.io/rhacs-eng/grpc-server:$(TAG) \
 		integration-tests/mock-grpc-server/image
 
+.PHONY: mock-grpc-server-image-push
+mock-grpc-server-image-push:
+	cp -R bin integration-tests/mock-grpc-server/image
+	docker buildx build --platform $(PLATFORM) --push \
```
Comment (Stringy) [inline Makefile:659]: Will this just build and push for amd64 currently? I think we're only building multi-arch binaries but not multi-arch images

Comment (kcrane) [reply Makefile:659]: As mentioned [here](https://github.com/stackrox/stackrox/pull/7482#discussion_r1297230267) only mock-grpc-server-image-push can be multi-arch.  I left the default PLATFORM as x86 in order to build the binaries and push the containers as multi-arch the PLATFORM needs to set in the environment.  I changed the build.yaml to refect this (NOTE: I'm not sure how to test build.yaml... it is untested).

Comment (kcrane) [reply Makefile:659]: Just noticed I should probably add a target dependency for push to ensure we have binaries:

ie. 
.PHONY: mock-grpc-server-image-push
mock-grpc-server-image-push: mock-grpc-server-build

Comment (Stringy) [reply Makefile]: build.yaml is running for this PR and appears to be green currently: https://github.com/stackrox/stackrox/actions/runs/5956150482/job/16156309812?pr=7482

**Review by Molter73 — APPROVED** [review]
LGTM!

**Pull request merged**